src/xdpinstaller/scripts/xdp-setup.ps1

```diff
@@ -0,0 +1,226 @@
+<#
+
+.SYNOPSIS
+This script installs or uninstalls various XDP components.
+
+.PARAMETER Install
+    Specifies an XDP component to install.
+
+.PARAMETER Uninstall
+    Attempts to uninstall all XDP components.
+
+#>
+
+param (
+    
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("", "xdp")]
+    [string]$Install = "",
+
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("", "xdp")]
+    [string]$Uninstall = ""
+)
+
+Set-StrictMode -Version 'Latest'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Global paths.
+$SystemFolder = [Environment]::SystemDirectory
+$XdpSys = "$SystemFolder\drivers\xdp.sys"
+$XdpInf = "$SystemFolder\drivers\xdp.inf"
+$XdpCat = "$SystemFolder\drivers\xdp.cat"
+$LogsDir = "$SystemFolder\Logs"
+
+# Set the temporary working directory and testore it on exit.
+Push-Location -Path $SystemFolder
+
+# Helper to reboot the machine.
+function Uninstall-Failure {
```

Comment:
This function won't actually work as is, because `$LiveKD` isn't set, nor is it going to be present on the target machine. I don't think we should have this as part of the uninstaller anyways for prod. It _should_ just always work. Let's remove this.

Reply:
We can mess around with this after the fact

Reply:
If MSI has a standard mechanism to trace errors, yeah that would be good.

Reply:
The MSI will log anything that is traced.